### ISSUE TITLE
Phase 1: Extract pure prepare_batch (no-op refactor, pinned by tests)

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#1739]
# Phase 1: Extract pure `prepare_batch` (no-op refactor, pinned by tests)

> **Type**: Phase · **Parent Epic**: #1738 · **Domain**: training · **Milestone**: training v1.0.0
> **Design doc**: `lance-dataloader-migration.md` §4, §8, §9
> **Blocked by**: — · **Blocks**: Phase 2, Phase 4

## Description

Pull the per-batch logic out of `VSTDataset.__getitem__` into a standalone **pure
function** `prepare_batch`, and have the *current* `__getitem__` call it so
behavior is byte-identical. This is deliberately a no-op refactor: it touches the
only thing in the migration that affects batch *semantics*, isolates it, and pins
it with a test. Every later phase is then pure plumbing that cannot silently
change what a batch contains.

This phase ships alone. After it, both the h5 and Lance paths route through the
same function, and batch semantics are frozen for the rest of the epic.

## Scope / tasks

- [ ] **Task 1.1 — Extract `prepare_batch`.** New pure function (suggested home:
  `src/synth_setter/data/batch.py` or a module-level function in
  `surge_datamodule.py`):

  ```python
  def prepare_batch(
      raw: dict[str, np.ndarray | None],   # {"param_array", "mel_spec"?, "music2latent"?, "audio"?}
      *,
      mean: np.ndarray | None,
      std: np.ndarray | None,
      rescale_params: bool,
      ot: bool,
      generator: torch.Generator,
  ) -> dict[str, torch.Tensor | None]:      # {"mel_spec","m2l","params","noise","audio"}
      ...
  ```

  It must own exactly today's per-batch steps, in the same order
  (`surge_datamodule.py` L236–272): mel normalization `(mel - mean) / std`; param
  rescale `[0,1] → [-1,1]`; `noise = randn_like(params)`; OT/Hungarian match
  (`_hungarian_match(noise, params, mel, audio)`); `.contiguous()` on outputs;
  `None` passthrough for unread features.

- [ ] **Task 1.2 — Thread an explicit `generator`.** Replace bare
  `torch.randn_like(...)` (and any RNG inside the OT path) with calls that accept
  a `torch.Generator`, so randomness is reproducible and pinnable. The current
  `__getitem__` constructs/holds a generator and passes it in; default seeding
  must reproduce today's distribution (document the seed source).

- [ ] **Task 1.3 — Rewire `__getitem__` to call `prepare_batch`.** The dataset
  still does its per-column reads exactly as today, assembles `raw`, then calls
  `prepare_batch(raw, ...)`. Samplers, `batch_size=None`, fake mode, and
  `repeat_first_batch` are untouched in this phase.

## Out of scope

- No Lance changes, no new dataset class, no collate_fn yet (that's Phase 2 —
  `prepare_batch` *becomes* the collate_fn there, unchanged).
- No sampler/dataloader changes.

## Testing — detailed

**Type: unit / pinning (the load-bearing test of this phase).**

- [ ] `test_prepare_batch_is_pure_and_pinned`
  - Build a fixed `raw` dict from small deterministic NumPy arrays (e.g. shapes
    `params=(8,N)`, `mel=(8,2,128,401)`), fixed `mean`/`std`.
  - Call `prepare_batch(raw, mean, std, rescale_params=True, ot=True,
    generator=torch.Generator().manual_seed(0))`.
  - Assert exact equality (`torch.equal` / `assert_allclose` with `atol=0`,
    `rtol=0` where math is exact; tight tolerance only where float reductions in
    OT make bit-exactness fragile — justify any tolerance in a comment) against a
    **golden** captured from the pre-refactor `__getitem__` on the same inputs and
    seed. Commit the golden as a small `.npz` fixture or generate it from a frozen
    reference function in the test.
  - Assert outputs are `.contiguous()` and dtype `float32`.
  - Assert purity: calling twice with the same `generator` seed yields identical
    output; calling with no mutation of `raw` leaves `raw` unchanged (no in-place
    writes to inputs).

- [ ] `test_prepare_batch_feature_flags`
  - Parametrize over the read-flag combinations the model uses:
    `(read_mel, read_m2l, read_audio)` ∈ {`(T,F,F)`, `(F,T,F)`, `(T,F,T)`}.
  - Assert the returned dict has the right keys non-`None`/`None`, and that
    `mel_spec`/`m2l`/`audio` are `None` exactly when their flag is off.
  - Assert mel normalization is applied iff both `mean` and `std` are set.

- [ ] `test_prepare_batch_rescale_toggle`
  - `rescale_params=False` leaves params in `[0,1]`; `True` maps to `[-1,1]`
    (check min/max bounds and that the transform is exactly `x*2-1`).

- [ ] `test_prepare_batch_ot_matches_reference`
  - With `ot=True`, assert the noise↔params pairing equals the reference
    `_hungarian_match` output for the seeded input (this is what guarantees we
    didn't reorder/realign rows during extraction).
  - With `ot=False`, assert noise/params/mel/audio pass through unpermuted.

**Type: regression (cheap guard).**

- [ ] `test_getitem_unchanged_after_extraction`
  - Run the *post-refactor* `VSTDataset.__getitem__` against a tiny real or
    synthetic shard for a fixed index + seed and compare to a golden captured
    from the pre-refactor implementation. This proves the rewire in Task 1.3 is a
    true no-op end-to-end (not just at the function boundary).

**What CI proves automatically vs. what these add.** CI already runs the suite +
lint; it does *not* know batch semantics are preserved. These tests are the
contract that the extraction changed structure but not output. Capture in the PR
"Test plan": the golden source (how it was generated), and that exactness holds
under fixed seed.

**Deliberately not tested here:** dataloader/sampler behavior and Lance reads
(Phase 2); performance (Phase 4 benchmark).

## Acceptance criteria

- `prepare_batch` exists as a pure, generator-seeded function; `__getitem__`
  delegates to it.
- All pinning/feature-flag/rescale/OT tests pass; the end-to-end `__getitem__`
  regression golden matches.
- No change to samplers, dataloaders, fake mode, or `repeat_first_batch`.
- PR follows Why → What changed → Test plan; references this phase + epic.
